diff --git a/libs/seiscomp3/mt/workspaceitem.cpp b/libs/seiscomp3/mt/workspaceitem.cpp
index 8a52724..8f61c14 100644
--- a/libs/seiscomp3/mt/workspaceitem.cpp
+++ b/libs/seiscomp3/mt/workspaceitem.cpp
@@ -39,6 +39,7 @@ using namespace std;
 using namespace Seiscomp::Core;
 
 #define TWO_PI (M_PI*2)
+#define IIR_BANDPASS ButterworthBandpass
 
 
 namespace Seiscomp {
@@ -1414,7 +1415,7 @@ bool WorkspaceItem::applyFilter(Component c, const FilterPeriods &periods) {
 		Filter *filter = NULL;
 
 		if ( periods.first > 0 && periods.second > 0 )
-			filter = new Math::Filtering::IIR::ButterworthHighLowpass<float>(4, 1.0/periods.second, 1.0/periods.first);
+			filter = new Math::Filtering::IIR::IIR_BANDPASS<float>(4, 1.0/periods.second, 1.0/periods.first);
 		else if ( periods.first > 0 )
 			filter = new Math::Filtering::IIR::ButterworthLowpass<float>(4, 1.0/periods.first);
 		else if ( periods.second > 0 )
@@ -1456,7 +1457,7 @@ bool WorkspaceItem::applyFilterGF(const FilterPeriods &periods) {
 
 			Filter *filter = NULL;
 			if ( periods.first > 0 && periods.second > 0 )
-				filter = new Math::Filtering::IIR::ButterworthHighLowpass<float>(4, 1.0/periods.second, 1.0/periods.first);
+				filter = new Math::Filtering::IIR::IIR_BANDPASS<float>(4, 1.0/periods.second, 1.0/periods.first);
 			else if ( periods.first > 0 )
 				filter = new Math::Filtering::IIR::ButterworthLowpass<float>(4, 1.0/periods.first);
 			else if ( periods.second > 0 )
