diff --git a/src/trunk/apps/tools/scimport/filter.cpp b/src/trunk/apps/tools/scimport/filter.cpp
index 774c3ec..d356c6a 100644
--- a/src/trunk/apps/tools/scimport/filter.cpp
+++ b/src/trunk/apps/tools/scimport/filter.cpp
@@ -24,6 +24,7 @@ namespace Applications
 std::map<std::string, Filter*> Filter::_Registry;
 std::map<std::string, Filter::Operator> Filter::_Operators;
 bool OriginFilter::_wasAccepted = false;
+bool MagnitudeFilter::_wasAccepted = false;
 // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
 
 
@@ -37,6 +38,7 @@ IMPLEMENT_MESSAGE_FILTER(ArrivalFilter)
 IMPLEMENT_MESSAGE_FILTER(EventFilter)
 IMPLEMENT_MESSAGE_FILTER(StationMagnitudeFilter)
 IMPLEMENT_MESSAGE_FILTER(MagnitudeFilter)
+IMPLEMENT_MESSAGE_FILTER(StationMagnitudeContributionFilter)
 IMPLEMENT_MESSAGE_FILTER(WaveformQualityFilter)
 // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
 
diff --git a/src/trunk/apps/tools/scimport/filter.h b/src/trunk/apps/tools/scimport/filter.h
index 2bdb279..fce076e 100644
--- a/src/trunk/apps/tools/scimport/filter.h
+++ b/src/trunk/apps/tools/scimport/filter.h
@@ -23,6 +23,7 @@
 #include <seiscomp3/datamodel/event.h>
 #include <seiscomp3/datamodel/stationmagnitude.h>
 #include <seiscomp3/datamodel/magnitude.h>
+#include <seiscomp3/datamodel/stationmagnitudecontribution.h>
 #include <seiscomp3/datamodel/waveformquality.h>
 #include <seiscomp3/datamodel/utils.h>
 #include <seiscomp3/core/strings.h>
@@ -717,6 +718,8 @@ class MagnitudeFilter : public Filter
 	DEFINE_MESSAGE_FILTER( DataModel::Magnitude::ClassName(), MagnitudeFilter )
 
 public:
+	static bool WasAccepted() { return _wasAccepted; }
+
 	virtual bool init(Client::Application &app) {
 		try {
 			_type = app.configGetString("filter.magnitude.type");
@@ -733,13 +736,55 @@ public:
 		if ( magnitude == NULL )
 			return false;
 
-		if ( _type )
-			if ( *_type != magnitude->type() )
-				return false;
+		_wasAccepted = true;
+
+		if ( _type ) {
+			if ( *_type != magnitude->type() ) {
+				_wasAccepted = false;
+			}
+		}
+
+		SEISCOMP_DEBUG("%s %s %s Type: %s AgencyID: %s",
+		               _wasAccepted?"+":"-",
+		               magnitude->className(), magnitude->publicID().c_str(),
+		               magnitude->type().c_str(), objectAgencyID(magnitude).c_str());
+
+		return _wasAccepted;
+	}
+
+private:
+	static bool      _wasAccepted;
+	OPT(std::string) _type;
+};
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+
+
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+class StationMagnitudeContributionFilter : public Filter
+{
+	DEFINE_MESSAGE_FILTER( DataModel::StationMagnitudeContribution::ClassName(), StationMagnitudeContributionFilter )
 
+public:
+	virtual bool init(Client::Application &app) {
 		return true;
 	}
 
+	virtual bool filterImpl(Core::BaseObject *object) {
+		DataModel::StationMagnitudeContribution *magref = DataModel::StationMagnitudeContribution::Cast(object);
+		if ( magref == NULL )
+			return false;
+
+		SEISCOMP_DEBUG("%s %s StationMagnitude: %s",
+		               MagnitudeFilter::WasAccepted()?"+":"-",
+		               magref->className(), magref->stationMagnitudeID().c_str());
+
+		// Only pass station magnitude contributions if the last magnitude was accepted
+		return MagnitudeFilter::WasAccepted();
+	}
+
 private:
 	OPT(std::string) _type;
 };
diff --git a/src/trunk/apps/tools/scimport/import.cpp b/src/trunk/apps/tools/scimport/import.cpp
index 9ca6c80..e88d6dd 100644
--- a/src/trunk/apps/tools/scimport/import.cpp
+++ b/src/trunk/apps/tools/scimport/import.cpp
@@ -76,6 +76,9 @@ bool Import::init() {
 	if (commandline().hasOption("test"))
 		_test = true;
 
+	try { _filter = configGetBool("useFilters"); }
+	catch ( ... ) {}
+
 	if (commandline().hasOption("no-filter"))
 		_filter = false;
 

